nemo/utils/data_utils.py

```diff
@@ -0,0 +1,312 @@
+
+    if cache_shared == 0:
+        return False
+    elif cache_shared == 1:
+        return True
+    else:
+        raise ValueError(f'Unexpected value of env {constants.NEMO_ENV_DATA_STORE_CACHE_SHARED}')
+
+
+def ais_cache_base() -> str:
+    """Return path to local cache for AIS.
+    """
+    override_dir = os.environ.get(constants.NEMO_ENV_DATA_STORE_CACHE_DIR, "")
+    if override_dir == "":
+        cache_dir = resolve_cache_dir().as_posix()
+    else:
+        cache_dir = pathlib.Path(override_dir).resolve().as_posix()
+
+    if cache_dir.endswith(NEMO_VERSION):
+        # Prevent re-caching dataset after upgrading NeMo
+        cache_dir = os.path.dirname(cache_dir)
+    return os.path.join(cache_dir, 'ais')
+
+
+def ais_endpoint() -> str:
+    """Get configured AIS endpoint.
+    """
+    return os.getenv('AIS_ENDPOINT')
+
+
+def bucket_and_object_from_uri(uri: str) -> Tuple[str, str]:
+    """Parse a path to determine bucket and object path.
+
+    Args:
+        uri: Full path to an object on an object store
+
+    Returns:
+        Tuple of strings (bucket_name, object_path)
+    """
+    if not is_datastore_path(uri):
+        raise ValueError(f'Provided URI is not a valid store path: {uri}')
+    uri_parts = pathlib.PurePath(uri).parts
+    bucket = uri_parts[1]
+    object_path = pathlib.PurePath(*uri_parts[2:])
+
+    return str(bucket), str(object_path)
+
+
+def ais_endpoint_to_dir(endpoint: str) -> str:
+    """Convert AIS endpoint to a valid dir name.
+    Used to build cache location.
+
+    Args:
+        endpoint: AIStore endpoint in format https://host:port
+    
+    Returns:
+        Directory formed as `host/port`.
+    """
+    if not endpoint.startswith('http://'):
+        raise ValueError(f'Unexpected format for ais endpoint: {endpoint}')
+
+    endpoint = endpoint.replace('http://', '')
+    host, port = endpoint.split(':')
+    return os.path.join(host, port)
+
+
+def ais_binary() -> str:
+    """Return location of `ais` binary.
+    """
+    path = shutil.which('ais')
+
+    if path is not None:
+        logging.debug('Found AIS binary at %s', path)
+        return path
+
+    logging.warning('AIS binary not found with `which ais`.')
+
+    # Double-check if it exists at the default path
+    default_path = '/usr/local/bin/ais'
+    if os.path.isfile(default_path):
+        logging.info('ais available at the default path: %s', default_path)
+        return default_path
+    else:
+        raise RuntimeError(f'AIS binary not found.')
+
+
+def datastore_path_to_local_path(store_path: str) -> str:
+    """Convert a data store path to a path in a local cache.
+
+    Args:
+        store_path: a path to an object on an object store
+
+    Returns:
+        Path to the same object in local cache.
+    """
+    if store_path.startswith('ais://'):
+        endpoint = ais_endpoint()
+        if endpoint is None:
+            raise RuntimeError(f'AIS endpoint not set, cannot resolve {path}')
```

Comment:
```suggestion
            raise RuntimeError(f'AIS endpoint not set, cannot resolve {store_path}')
```
@titu1994 - I strongly suggest revisiting our CI setup to make sure such errors are caught by linter

Reply:
<img width="1415" alt="image" src="https://user-images.githubusercontent.com/3048602/206810923-8be5f1ec-3dfb-44e5-a9fc-f6eb7c9c4dd1.png">
Its managed by Github CodeQL - https://github.com/NVIDIA/NeMo/pull/5443 in the autocast() function of eval_beam_search.